crates/iceberg/src/arrow/reader.rs

```diff
@@ -351,19 +383,26 @@ impl ArrowReader {
                 true
             });
 
-            if column_map.len() != field_ids.len() {
+            if column_map.len() != leaf_field_ids.len() {
+                let miss_field = leaf_field_ids
```

Comment:
This var should be called `missing_fields` I think, and similarly in the associated `with_context` (I love that these contexts are getting added by the way 😄 )